src/payment/strategies/zip/zip-payment-strategy.ts

```diff
@@ -80,6 +81,10 @@ export default class ZipPaymentStrategy implements PaymentStrategy {
                                     return resolve(checkoutId);
                                 }
 
+                                if (state === ZipModalEvent.CheckoutDeclined) {
+                                    return reject(new StandardError('Unfortunately your application was declined. Please select another payment method.'));
```

Comment:
It might be a good idea to define a more specific error type. i.e.: `PaymentDeclinedError` instead of generic `StandardError`. This way, the SDK user can simply check the `type` and react to it. Otherwise, they will need to check the error message itself, which is not very useful.

PS: I think currently `StandardError` is getting abused so I'm thinking about making it an abstract class instead in the future.